Library/Formula/mlkit.rb

```diff
@@ -0,0 +1,23 @@
+class Mlkit < Formula
+  desc "Compiler for the Standard ML programming language"
+  homepage "https://melsman.github.io/mlkit"
+  url "https://github.com/melsman/mlkit/archive/mlkit-4.3.9.tar.gz"
+  sha256 "3c6adbeb9a85f7b3586d0961fd3b170ff31e09fa0ff12889b76b9ceb459059c4"
+
+  depends_on "autoconf" => :build
+  depends_on "mlton" => :build
+  depends_on "gmp"
+
+  def install
+    system "./autobuild; true"
+    system "./configure", "--prefix=#{prefix}"
+    ENV.permit_arch_flags
```

Comment:
For the mlton-compilation of sml-code, no arch tags are used. Thus, on a
32-bit machine, both the mlton-compiled binary (the mlkit compiler) and the
32-bit native code generated by the mlkit compiler will be running 32-bit
code.

The backend for the mlkit is not easily ported to run on 64-bit
architectures, which is a major engineering task and which is currently not
a planned activity. I hope that this limitation is not a show-stopper for
accepting the mlkit formula in homebrew.

On Sun, Mar 13, 2016 at 5:24 PM, Mike McQuaid <notifications@github.com>
wrote:

> In Library/Formula/mlkit.rb
> <https://github.com/Homebrew/homebrew/pull/49933#discussion_r55939421>:
>
> > @@ -0,0 +1,23 @@
> > +class Mlkit < Formula
> > +  desc "Compiler for the Standard ML programming language"
> > +  homepage "https://melsman.github.io/mlkit"
> > +  url "https://github.com/melsman/mlkit/archive/mlkit-4.3.9.tar.gz"
> > +  sha256 "3c6adbeb9a85f7b3586d0961fd3b170ff31e09fa0ff12889b76b9ceb459059c4"
> > +
> > +  depends_on "autoconf" => :build
> > +  depends_on "mlton" => :build
> > +  depends_on "gmp"
> > +
> > +  def install
> > +    system "./autobuild; true"
> > +    system "./configure", "--prefix=#{prefix}"
> > +    ENV.permit_arch_flags
>
> How does this behave on a 32-bit only machine? Why does it generate 32-bit
> code on a 64-bit machine?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/49933/files#r55939421>.
>